### ISSUE TITLE
Creating PDF.

### ISSUE THREAD

**Issue by Krosovok** [Krosovok/MorningPaperRepo#5]
What do we know about creating PDF files? Shandress, report.

**Comment by Shandress** [comment]
There are a decent amount of C# libraries that work with PDF files, i.e. creating a PDF from scratch, adding chapters, sentences etc. Such libraries, like iTextSharp, MigraDoc, PDFSharp and others also support formatting, so we are able to set a Font, for example. One of the most interesting things here is that we can add pages/chapters to a pdf from an html code (I don't know how exactly it works, but is surely an option). I also found a "compilation" of such dll's on CodeProject that will give us a possibility to create pdf files from html. And I'm currently testing these libraries to ensure that I know how to use them properly and to full extent.
I'd like to hear your ideas about what exactly do we need when it comes to pdf files. e.g. do we also need to create PDF's from text, XML, Word docs whatever? Is the Idea html code ==> good or should I try finding something that can do a web page ==> pdf conversion or something else I haven't thought about. To cut it short, post your ideas to keep me focused in the right direction.

**Comment by Krosovok** [comment]
I suppos that we will store for each site a "path" (I don't know how) to a tag which contains article. And then, we will convert content of this tag to our PDF article. That means that we need only HTML ==> PDF convertation as long as we don't need the whole pages with advertising and other useles parts. We need only information. Maybe some images but not much more.

**Comment by Shandress** [comment]
I looked further into parsing html and creating PDFs and that's what I've accomplished so far:
1. Created classes that represent a piece of formatted text and an image, that can be added to a pdf document, both implement the IArticleElement interface. As for the text element, a font can be set to make it formatted, image - nothing is done except the constructor. Both elements can only be created from the HtmlNode object that we get using the HtmlAgilityPack. 
2. Tested text element class a little.
3. Encountered some problems, e.g. there's only so much info regarding formatting that we can get from html, to make the text formatted we need to parse CSS, I took a brief look at the issue and think it's possible to do, I just don't know how to get the css file of the page to work with.
That's what I've done so far. Any help would be appreciated.
